src/qinfer/derived_models.py

```diff
@@ -599,6 +602,220 @@ def update_timestep(self, modelparams, expparams):
         steps = steps.transpose((0, 2, 1))
         
         return modelparams[:, :, np.newaxis] + steps
+        
+class GaussianRandomWalkModel(DerivedModel):
+    r"""
+    Model such that after each time step, a random perturbation is 
+    added to each model parameter vector according to a 
+    zero-mean gaussian distribution.
+    
+    The :math:`n\times n` covariance matrix of this distribution is 
+    either fixed and known, or its entries are treated as unknown, 
+    being appended to the model parameters.
+    For diagonal covariance matrices, :math:`n` parameters are added to the model 
+    storing the square roots of the diagonal entries of the covariance matrix.
+    For dense covariance matrices, :math:`n(n+1)/2` parameters are added to 
+    the model, storing the entries of the lower triangular portion of the
+    Cholesky factorization of the covariance matrix.
+    
+    :param Model underlying_model: Model representing the likelihood with no
+        random walk added.
+    :param random_walk_idxs: A list of model parameter indeces to add the 
```

Comment:
Hmm, yes, my problem with using `np.s_[:]` for `random_walk_idxs` is that this parameter is used to slice a subset of the underlying model parameters, but `np.s_[:]` would refer to both those and the new ones added by `GaussianRandomWalkModel`. Therefore I think we need a keyword or equivalent, since on creation we don't yet know how many extra parameters there will be.

Though it it not tested, I think the user can still use fancy indices for `np.s_[:]` -- I will add this to the unit tests.